RxCocoa/Common/DelegateProxy.swift

```diff
@@ -201,18 +203,45 @@ open class DelegateProxy : _RXDelegateProxy {
     ///
     /// - parameter object: Object that can have assigned delegate proxy.
     /// - parameter proxy: Delegate proxy object to assign to `object`.
-    open class func assignProxy(_ proxy: AnyObject, toObject object: AnyObject) {
-        precondition(proxy.isKind(of: self.classForCoder()))
-       
+    open class func assignProxy(_ proxy: Delegate, toObject object: ParentObject) {
         objc_setAssociatedObject(object, self.delegateAssociatedObjectTag(), proxy, .OBJC_ASSOCIATION_RETAIN)
     }
     
+    
+    /// Returns designated delegate property for object.
+    ///
+    /// Objects can have multiple delegate properties.
+    ///
+    /// Each delegate property needs to have it's own type implementing `DelegateProxyType`.
+    ///
+    /// It's abstract method.
+    ///
+    /// - parameter object: Object that has delegate property.
+    /// - returns: Value of delegate property.
+    open class func currentDelegateFor(_ object: ParentObject) -> Delegate? {
```

Comment:
Is there some reason we are conforming to `DelegateProxy` to `DelegateProxyBase`. It seems to me we aren't getting anything from that and using abstract methods in Swift is usually avoided.

Yes, we use them in a couple of places, but we don't need here IMHO.

Reply:
It seems to me that we don't need this method defined here.